src/test/codegen/simd-intrinsic/simd-intrinsic-generic-extract-insert.rs

```diff
@@ -0,0 +1,47 @@
+// compile-flags: -C no-prepopulate-passes
+
+#![crate_type = "lib"]
+
+#![feature(repr_simd, platform_intrinsics, min_const_generics)]
+#![allow(non_camel_case_types)]
+
+#[repr(simd)]
+#[derive(Copy, Clone)]
+pub struct M(pub f32, pub f32, pub f32, pub f32);
+
+#[repr(simd)]
+#[derive(Copy, Clone)]
+pub struct S<const N: usize>([f32; N]);
```

Comment:
Can you also try `pub struct S2<T: SomeTrait>([f32; T::SOME_ASSOC_CONST]);`? It's similar, but not exactly the same thing

Reply:
Hmm, this gives me the `error: constant expression depends on a generic parameter`.

Is the idea to add a compile-fail test to catch ourselves if that starts being supported?

Reply:
Yea, I think in this case it would be good to add a compile-fail test as a canary.